### ISSUE TITLE
module_defaults for yum do not get picked up when invoked via package

### ISSUE THREAD

**Issue by opoplawski** [ansible/ansible#57482]
##### SUMMARY
module_defaults for yum do not get picked up when invoked via package.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module_defaults

##### ANSIBLE VERSION
```paste below
ansible 2.8.0
  config file = /export/home/orion-admin/ansible-boulder/ansible.cfg
  configured module search path = [u'/export/home/orion-admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Apr  8 2019, 14:44:40) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/export/home/orion-admin/ansible-boulder/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/export/home/orion-admin/ansible-boulder/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=60s -o ForwardX11=no -o ForwardAgent=no
DEFAULT_BECOME(/export/home/orion-admin/ansible-boulder/ansible.cfg) = True
DEFAULT_CALLBACK_WHITELIST(/export/home/orion-admin/ansible-boulder/ansible.cfg) = [u'profile_tasks']
DEFAULT_FORKS(/export/home/orion-admin/ansible-boulder/ansible.cfg) = 25
DEFAULT_GATHERING(/export/home/orion-admin/ansible-boulder/ansible.cfg) = smart
DEFAULT_HOST_LIST(/export/home/orion-admin/ansible-boulder/ansible.cfg) = [u'/export/home/orion-admin/ansible-boulder/inventory']
DEFAULT_ROLES_PATH(/export/home/orion-admin/ansible-boulder/ansible.cfg) = [u'/export/home/orion-admin/ansible-boulder/roles']
DEFAULT_TIMEOUT(/export/home/orion-admin/ansible-boulder/ansible.cfg) = 20
PERSISTENT_CONNECT_TIMEOUT(/export/home/orion-admin/ansible-boulder/ansible.cfg) = 30
RETRY_FILES_ENABLED(/export/home/orion-admin/ansible-boulder/ansible.cfg) = False
```

##### OS / ENVIRONMENT
EL7

##### STEPS TO REPRODUCE

```yaml
---
- hosts: localhost
  module_defaults:
    yum:
      lock_timeout: 120
  tasks:
    - package:
        name: yum
        state: present
    - yum:
        name: yum
        state: present
```

##### EXPECTED RESULTS

lock_timeout is set to 120 for both tasks

##### ACTUAL RESULTS

```paste below
TASK [package] *************************************************************************************************************************************************************
task path: /export/home/orion-admin/ansible-boulder/module_defaults.yml:7
Thursday 06 June 2019  08:11:13 -0600 (0:00:03.361)       0:00:03.478 ********* 
Running yum
Using module file /usr/lib/python2.7/site-packages/ansible/modules/packaging/os/yum.py
Pipelining is enabled.
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: orion-admin
<127.0.0.1> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-hterphlkdkurpiicergmqnjyvgknrbjw ; /usr/bin/python2'"'"' && sleep 0'
ok: [localhost] => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "allow_downgrade": false, 
            "autoremove": false, 
            "bugfix": false, 
            "conf_file": null, 
            "disable_excludes": null, 
            "disable_gpg_check": false, 
            "disable_plugin": [], 
            "disablerepo": [], 
            "download_dir": null, 
            "download_only": false, 
            "enable_plugin": [], 
            "enablerepo": [], 
            "exclude": [], 
            "install_repoquery": true, 
            "install_weak_deps": true, 
            "installroot": "/", 
            "list": null, 
            "lock_timeout": 0, 
            "name": [
                "yum"
            ], 
            "releasever": null, 
            "security": false, 
            "skip_broken": false, 
            "state": "present", 
            "update_cache": false, 
            "update_only": false, 
            "use_backend": "auto", 
            "validate_certs": true
        }
    }, 
    "msg": "", 
    "rc": 0, 
    "results": [
        "yum-3.4.3-161.sl7.noarch providing yum is already installed"
    ]
}

TASK [yum] *****************************************************************************************************************************************************************
task path: /export/home/orion-admin/ansible-boulder/module_defaults.yml:10
Thursday 06 June 2019  08:11:14 -0600 (0:00:01.152)       0:00:04.631 ********* 
Running yum as the backend for the yum action plugin
Using module file /usr/lib/python2.7/site-packages/ansible/modules/packaging/os/yum.py
Pipelining is enabled.
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: orion-admin
<127.0.0.1> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-vkzlzxdlhpeovmkjkoxcvxvcdidbaoeo ; /usr/bin/python2'"'"' && sleep 0'
ok: [localhost] => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "allow_downgrade": false, 
            "autoremove": false, 
            "bugfix": false, 
            "conf_file": null, 
            "disable_excludes": null, 
            "disable_gpg_check": false, 
            "disable_plugin": [], 
            "disablerepo": [], 
            "download_dir": null, 
            "download_only": false, 
            "enable_plugin": [], 
            "enablerepo": [],
            "exclude": [],
            "install_repoquery": true,
            "install_weak_deps": true,
            "installroot": "/",
            "list": null,
            "lock_timeout": 120,
            "name": [
                "yum"
            ],
            "releasever": null,
            "security": false,
            "skip_broken": false,
            "state": "present",
            "update_cache": false,
            "update_only": false,
            "use_backend": "auto",
            "validate_certs": true
        }
    },
    "msg": "",
    "rc": 0,
    "results": [
        "yum-3.4.3-161.sl7.noarch providing yum is already installed"
    ]
}
```

**Comment by sivel** [comment]
This should be resolved in https://github.com/ansible/ansible/pull/56400 which is waiting to be backported into 2.8

**Comment by sivel** [comment]
!component =lib/ansible/plugins/action/package.py

**Comment by samdoran** [comment]
Fixed in 674a57c3fa9abe1ccc6a22d2d5cff4ba8a8ba0b8
